modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/QuantumToolkit.java

```diff
@@ -259,7 +259,25 @@ boolean get() {
          */
         shutdownHook = new Thread("Glass/Prism Shutdown Hook") {
             @Override public void run() {
-                dispose();
+                // Run dispose in a background thread and wait for up to
+                // 5 seconds for it to finish. If it doesn't, then throw an
```

Comment:
is 5 seconds a good timeout value?

could it be shorter?
is it possible for the normal shutdown on a real system take more than 5 seconds?

Reply:
I think 5 seconds is a reasonable timeout value. I wouldn't want it to be any shorter nor too much longer. Normally, this is something that would take a few milliseconds.

FWIW, we have a similar 5 second timeout in the (macOS-specific) initialization code that waits for app activation.